### DISCUSSION TITLE
Transcribing video and changing output

### DISCUSSION THREAD

**Discussion by kanishknishar** [openai/whisper#820] [Q&A]
I am very new to programming and don't know Python so trying to wrap my head around the instructions provided for Whisper is very difficult. I had two questions:
1) I want to change the output of Whisper to only vtt or srt files. How do I do that?
2) Is there a way to automate transcription of video files via the command line? Currently I am using Audacity to import video files and export it as FLAC.
3. How do I make the FP16 not supported error go away permanently? Currently I have to supply the argument --fp16 False every time.

I only have VSCode at the moment. No Jupyter notebook if that's necessary for this.

**Comment by kanishknishar** [comment]
I did find this tool: https://github.com/dsymbol/decipher

It convert the files to AAC by default it seems. Is there such a thing as because this is a third party tool it might not be as updated as using Whisper directly?

**Comment by glangford** [comment]
> I want to change the output of Whisper to only vtt or srt files. How do I do that?

There is no command line argument to omit .txt output currently. This is where text files are written (and further down VTT and SRT). 
https://github.com/openai/whisper/blob/28769fcfe50755a817ab922a7bc83483159600a9/whisper/transcribe.py#L313

> Is there a way to automate transcription of video files via the command line? Currently I am using Audacity to import video files and export it as FLAC.

whisper can accept containers of video+audio directly, since it uses ffmpeg under the hood. So if your video file is .mp4 for example, just use whisper on that. 

If for some reason that doesn't work, you can always extract audio in a script using ffmpeg, eg.
`ffmpeg -i input-video.avi -vn -acodec copy output-audio.aac`

> How do I make the FP16 not supported error go away permanently? Currently I have to supply the argument --fp16 False every time.

Keep using `--fp16 False`
Here is where the warning is generated
https://github.com/openai/whisper/blob/02aa851a4910201f0db56960064d7e121a01002c/whisper/transcribe.py#L78

**Comment by kanishknishar** [comment]
So if I want to remove txt and vtt, I just need to remove those lines of codes from my transcribe.py file?

**Comment by glangford** [comment]
> So if I want to remove txt and vtt, I just need to remove those lines of codes from my transcribe.py file?

Yes, or just comment out the parts you don't want (in case you want them back later) 👍

**Comment by kanishknishar** [comment]
`whisper can accept containers of video+audio directly, since it uses ffmpeg under the hood. So if your video file is .mp4 for example, just use whisper on that.`

I did feed it an MP4 file. It didn't work. It worked with decipher though.

**Comment by leeterboy** [comment]
--output_format srt txt
this works for me when put at the end using google colab.
